shell/platform/android/BUILD.gn

```diff
@@ -71,6 +71,13 @@ shared_library("flutter_shell_native") {
     "EGL",
     "GLESv2",
   ]
+
+
+  if (flutter_runtime_mode != "release") {
+    # This flag is needed so that the call to dladdr() in Dart's native symbol
+    # resolver can report good symbol information for the CPU profiler.
+    ldflags = [ "-rdynamic" ]
```

Comment:
Does it make sense to put this in the GN file that defines the link tool for all `.so`s? You are already adding the `enable_profiling` flag. So it should be available there.